File: ExcludedSamples/Reports/C1Report/VB/C1dView/C1dView.ZoomDialog.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports C1.Win.C1Preview

Public Class ZoomDialog

    Public Property PreviewPane As C1.Win.C1Preview.C1PreviewPane

    Private Property Updating As Boolean

    Public Sub New()

        ' This call is required by the designer.
        InitializeComponent()
        ''
        AddHandler Me.numZoom.ValueChanged, _
            Sub(s, e)
                If (Not Updating) Then
                    PreviewPane.ZoomFactor = CType(numZoom.Value / 100D, Double)
                    UpdateFromPreview()
                End If
            End Sub

        AddHandler Me.numCols.ValueChanged, _
            Sub(s, e)
                If (Not Updating) Then
                    PreviewPane.ZoomMode = ZoomModeEnum.WholePage
                    PreviewPane.Cols = CType(numCols.Value, Integer)
                    UpdateFromPreview()
                End If
            End Sub

        AddHandler Me.numRows.ValueChanged, _
            Sub(s, e)
                If (Not Updating) Then
                    PreviewPane.ZoomMode = ZoomModeEnum.WholePage
                    PreviewPane.Continuous = False
                    PreviewPane.Rows = CType(numRows.Value, Integer)
                    UpdateFromPreview()
                End If
            End Sub

        AddHandler Me.chkFacing.CheckedChanged, _
            Sub(s, e)
                If (Not Updating) Then
                    PreviewPane.ZoomMode = ZoomModeEnum.WholePage
                    PreviewPane.PageLayout = If(Me.chkFacing.Checked, PageLayoutEnum.Facing, PageLayoutEnum.Normal)
                    UpdateFromPreview()
                End If
            End Sub

        AddHandler Me.chkContinuous.CheckedChanged, _
            Sub(s, e)
                If (Not Updating) Then
                    PreviewPane.Continuous = Me.chkContinuous.Checked
                    UpdateFromPreview()
                End If
            End Sub

        AddHandler Me.rbt100.CheckedChanged, _
            Sub(s, e)
                If (Not Updating AndAlso rbt100.Checked) Then
                    PreviewPane.ZoomMode = ZoomModeEnum.ActualSize
                    UpdateFromPreview()
                End If
            End Sub

        AddHandler Me.rbtFitWindow.CheckedChanged, _
            Sub(s, e)
                If (Not Updating AndAlso rbtFitWindow.Checked) Then
                    PreviewPane.ZoomMode = ZoomModeEnum.WholePage
                    UpdateFromPreview()
                End If
            End Sub

        AddHandler Me.rbtFitWidth.CheckedChanged, _
            Sub(s, e)
                If (Not Updating AndAlso rbtFitWidth.Checked) Then
                    PreviewPane.ZoomMode = ZoomModeEnum.PageWidth
                    UpdateFromPreview()
                End If
            End Sub

        AddHandler Me.rbtFitTextWidth.CheckedChanged, _
            Sub(s, e)
                If (Not Updating AndAlso rbtFitTextWidth.Checked) Then
                    PreviewPane.ZoomMode = ZoomModeEnum.TextWidth
                    UpdateFromPreview()
                End If
            End Sub

        AddHandler Me.btnOk.Click, _
            Sub(s, e)
                Me.DialogResult = System.Windows.Forms.DialogResult.OK
                Me.Close()
            End Sub

    End Sub

    Protected Overrides Sub OnShown(e As System.EventArgs)
        UpdateLimits()
        UpdateFromPreview()
        MyBase.OnShown(e)
    End Sub

    Private Sub UpdateLimits()
        If (PreviewPane Is Nothing OrElse Updating) Then
            Return
        End If
        Try
            Updating = True
            numZoom.Minimum = CType(Math.Round(PreviewPane.MinZoomFactor * 100), Integer)
            numZoom.Maximum = CType(Math.Round(PreviewPane.MaxZoomFactor * 100), Integer)
        Finally
            Updating = False
        End Try
    End Sub

    Private Sub UpdateFromPreview()
        If (PreviewPane Is Nothing OrElse Updating) Then
            Return
        End If
        Try
            Updating = True
            numZoom.Value = CType(Math.Round(PreviewPane.ZoomFactor * 100), Integer)
            numCols.Value = PreviewPane.Cols
            numRows.Value = PreviewPane.Rows
            chkFacing.Checked = PreviewPane.PageLayout = C1.Win.C1Preview.PageLayoutEnum.Facing
            chkContinuous.Checked = PreviewPane.Continuous
            rbt100.Checked = PreviewPane.ZoomFactor = 1
            rbtFitWindow.Checked = PreviewPane.ZoomMode = C1.Win.C1Preview.ZoomModeEnum.WholePage
            rbtFitWidth.Checked = PreviewPane.ZoomMode = C1.Win.C1Preview.ZoomModeEnum.PageWidth
            rbtFitTextWidth.Checked = PreviewPane.ZoomMode = C1.Win.C1Preview.ZoomModeEnum.TextWidth
        Finally
            Updating = False
        End Try
    End Sub
End Class

File: ExcludedSamples/NetFramework/Reports/C1Report/Cs/C1dView/C1dView.ZoomDialog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using C1.Win.C1Preview;

namespace C1dView
{
    public partial class ZoomDialog : Form
    {
        public ZoomDialog()
        {
            InitializeComponent();

            this.numZoom.ValueChanged += (s, e) =>
                {
                    if (!Updating)
                    {
                        PreviewPane.ZoomFactor = (double)numZoom.Value / 100d;
                        UpdateFromPreview();
                    }
                };
            this.numCols.ValueChanged += (s, e) =>
                {
                    if (!Updating)
                    {
                        PreviewPane.ZoomMode = ZoomModeEnum.WholePage;
                        PreviewPane.Cols = (int)numCols.Value;
                        UpdateFromPreview();
                    }
                };
            this.numRows.ValueChanged += (s, e) =>
                {
                    if (!Updating)
                    {
                        PreviewPane.ZoomMode = ZoomModeEnum.WholePage;
                        PreviewPane.Continuous = false;
                        PreviewPane.Rows = (int)numRows.Value;
                        UpdateFromPreview();
                    }
                };
            this.chkFacing.CheckedChanged += (s, e) =>
                {
                    if (!Updating)
                    {
                        PreviewPane.ZoomMode = ZoomModeEnum.WholePage;
                        PreviewPane.PageLayout = this.chkFacing.Checked ? PageLayoutEnum.Facing : PageLayoutEnum.Normal;
                        UpdateFromPreview();
                    }
                };
            this.chkContinuous.CheckedChanged += (s, e) =>
                {
                    if (!Updating)
                    {
                        PreviewPane.Continuous = this.chkContinuous.Checked;
                        UpdateFromPreview();
                    }
                };
            this.rbt100.CheckedChanged += (s, e) =>
                {
                    if (!Updating && rbt100.Checked)
                    {
                        PreviewPane.ZoomMode = ZoomModeEnum.ActualSize;
                        UpdateFromPreview();
                    }
                };
            this.rbtFitWindow.CheckedChanged += (s, e) =>
                {
                    if (!Updating && rbtFitWindow.Checked)
                    {
                        PreviewPane.ZoomMode = ZoomModeEnum.WholePage;
                        UpdateFromPreview();
                    }
                };
            this.rbtFitWidth.CheckedChanged += (s, e) =>
                {
                    if (!Updating && rbtFitWidth.Checked)
                    {
                        PreviewPane.ZoomMode = ZoomModeEnum.PageWidth;
                        UpdateFromPreview();
                    }
                };
            this.rbtFitTextWidth.CheckedChanged += (s, e) =>
                {
                    if (!Updating && rbtFitTextWidth.Checked)
                    {
                        PreviewPane.ZoomMode = ZoomModeEnum.TextWidth;
                        UpdateFromPreview();
                    }
                };

            this.btnOk.Click += (s, e) =>
                {
                    this.DialogResult = System.Windows.Forms.DialogResult.OK;
                    this.Close();
                };
        }

        public C1.Win.C1Preview.C1PreviewPane PreviewPane { get; set; }

        private bool Updating { get; set; }

        protected override void OnShown(EventArgs e)
        {
            UpdateLimits();
            UpdateFromPreview();
            base.OnShown(e);
        }

        private void UpdateLimits()
        {
            if (PreviewPane == null || Updating)
                return;
            try
            {
                Updating = true;
                numZoom.Minimum = (int)Math.Round(PreviewPane.MinZoomFactor * 100);
                numZoom.Maximum = (int)Math.Round(PreviewPane.MaxZoomFactor * 100);
            }
            finally
            {
                Updating = false;
            }
        }

        private void UpdateFromPreview()
        {
            if (PreviewPane == null || Updating)
                return;
            try
            {
                Updating = true;
                numZoom.Value = (int)Math.Round(PreviewPane.ZoomFactor * 100);
                numCols.Value = PreviewPane.Cols;
                numRows.Value = PreviewPane.Rows;
                chkFacing.Checked = PreviewPane.PageLayout == C1.Win.C1Preview.PageLayoutEnum.Facing;
                chkContinuous.Checked = PreviewPane.Continuous;
                rbt100.Checked = PreviewPane.ZoomFactor == 1;
                rbtFitWindow.Checked = PreviewPane.ZoomMode == C1.Win.C1Preview.ZoomModeEnum.WholePage;
                rbtFitWidth.Checked = PreviewPane.ZoomMode == C1.Win.C1Preview.ZoomModeEnum.PageWidth;
                rbtFitTextWidth.Checked = PreviewPane.ZoomMode == C1.Win.C1Preview.ZoomModeEnum.TextWidth;
            }
            finally
            {
                Updating = false;
            }
        }
    }
}
